src/components/modals/RedelegateModal.tsx

```diff
@@ -0,0 +1,350 @@
+import { validateIOAmount } from '../forms/validation';
+import BaseModal from './BaseModal';
+import ReviewRedelegateModal from './ReviewRedelegateModal';
+import { REDELEGATION_FEE_TOOLTIP_TEXT } from '@src/constants';
+
+export type RedelegateModalProps = {
+  onClose: () => void;
+  sourceGateway: AoGatewayWithAddress;
+  vaultId?: string;
+  maxRedelegationStake: ARIOToken;
+};
+
+const RedelegateModal = ({
+  onClose,
+  sourceGateway,
+  vaultId,
+  maxRedelegationStake,
+}: RedelegateModalProps) => {
+  const walletAddress = useGlobalState((state) => state.walletAddress);
+  const ticker = useGlobalState((state) => state.ticker);
+
+  const { data: delegateStakes } = useDelegateStakes(walletAddress?.toString());
+
+  const [targetGateway, setTargetGateway] = useState<AoGatewayWithAddress>();
+  const [targetGatewayCurrentStake, setTargetGatewayCurrentStake] =
+    useState<number>();
+
+  const [amountToRedelegate, setAmountToRedelegate] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [isFormValid, setIsFormValid] = useState<boolean>(false);
+
+  const [showReviewRedelegateModal, setShowReviewRedelegateModal] =
+    useState(false);
+
+  const { data: redelegationFee } = useRedelegationFee(
+    walletAddress?.toString(),
+  );
+
+  const { data: gateways } = useGateways();
+  const [filteredGateways, setFilteredGateways] =
+    useState<AoGatewayWithAddress[]>();
+
+  useEffect(() => {
+    if (gateways) {
+      const filteredGateways = Object.entries(gateways)
+        .map(([address, gateway]) => {
+          return { ...gateway, gatewayAddress: address };
+        })
+        .filter(
+          (gateway) =>
+            gateway.status === 'joined' &&
+            gateway.settings.allowDelegatedStaking &&
+            gateway.gatewayAddress !== sourceGateway.gatewayAddress,
+        );
+      setFilteredGateways(filteredGateways);
+    }
+  }, [gateways, sourceGateway.gatewayAddress]);
+
+  useEffect(() => {
+    if (targetGateway && walletAddress) {
+      if (targetGateway.gatewayAddress === walletAddress.toString()) {
+        setTargetGatewayCurrentStake(
+          new mARIOToken(targetGateway.operatorStake).toARIO().valueOf(),
+        );
+      } else if (delegateStakes) {
+        const stake = delegateStakes.stakes.find(
+          (stake) => stake.gatewayAddress === targetGateway.gatewayAddress,
+        )?.balance;
+        setTargetGatewayCurrentStake(
+          new mARIOToken(stake ?? 0).toARIO().valueOf(),
+        );
+      }
+    }
+  }, [targetGateway, delegateStakes, walletAddress]);
+
+  const minDelegatedStake = targetGateway
+    ? new mARIOToken(targetGateway?.settings.minDelegatedStake)
+        .toARIO()
+        .valueOf()
+    : 10;
+  const minRequiredStakeToAdd =
+    (targetGatewayCurrentStake ?? 0) > 0 ? 1 : minDelegatedStake;
+
+  const validators = useMemo(
+    () => ({
+      redelegationAmount: validateIOAmount(
```

Comment:
Can do a find and replace for validateIOAmount

Reply:
done